lib/tapioca/compilers/dsl/active_record_columns.rb

```diff
@@ -347,6 +347,28 @@ def do_not_generate_strong_types?(constant)
           Object.const_defined?(:StrongTypeGeneration) &&
               !(constant.singleton_class < Object.const_get(:StrongTypeGeneration))
         end
+
+        def handle_unknown_type(column_type)
+          return "T.untyped" unless column_type < ActiveModel::Type::Value
+
+          lookup_return_type_of_method(column_type, :deserialize) ||
+            lookup_return_type_of_method(column_type, :cast) ||
+            lookup_arg_type_of_method(column_type, :serialize) ||
+            "T.untyped"
+        end
+
+        def lookup_return_type_of_method(column_type, method)
+          signature = T::Private::Methods.signature_for_method(column_type.instance_method(method))
+          return unless signature
+
+          signature.return_type.to_s if signature.return_type.to_s != "<VOID>" &&
+              signature.return_type.to_s != "<NOT-TYPED>"
+        end
```

Comment:
This is how I refactored  it, thought it was the easiest to read option. Let me know 👍